Dockerfile

```diff
@@ -16,18 +16,20 @@ RUN mkdir /app
 RUN pip3 install --upgrade pip
 RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
 RUN /root/.local/bin/poetry config virtualenvs.create false
-COPY pyproject.toml /app
-COPY poetry.lock /app
+COPY pyproject.toml poetry.lock /app/
 WORKDIR /app
-RUN /root/.local/bin/poetry install --no-root --extras "all"
+RUN  /root/.local/bin/poetry install --no-root --extras "all" \
+ &&  apt-get purge -y --auto-remove gcc 
 
 COPY src/ /app/src
 
 RUN pip3 install --use-feature=in-tree-build .
-# Install tabulate version that fixes column width wrapping. Cannot be added to pypi as a git dependency, so adding it here
-RUN pip3 install git+https://github.com/astanin/python-tabulate.git@b2c26bcb70e497f674b38aa7e29de12c0123708a#egg=tabulate
 
 COPY playbooks/ /etc/robusta/playbooks/defaults
 
 # -u disables stdout buffering https://stackoverflow.com/questions/107705/disable-output-buffering
-CMD [ "python3", "-u", "-m", "robusta.runner.main"]
\ No newline at end of file
+CMD [ "python3", "-u", "-m", "robusta.runner.main"]
+
+FROM dev as prd
```

Comment:
Due to the way you're inheriting from the previous stage here, we still carry the layer that contains `gcc` with us. GCC isn't present in the final image (because you uninstall it) but it is present in one of the previous layers which inflates the image size.

I think you have to either switch to a new base image (FROM newimage) which is clean or do the trick where you install and uninstall gcc on the same line